.github/workflows/ci.yaml

```diff
@@ -0,0 +1,82 @@
+name: CI
+
+on:
+  pull_request:
+    branches:
+      - main
+  push:
+    branches:
+      - main
+    tags: [v*]
+  workflow_dispatch:
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  lint:
+    name: Lint
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Run pre-commit
+        run: pipx run pre-commit run --all-files
+
+  test:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest, windows-latest, macos-latest]
+        python-version: ["3.9", "3.10", "3.11", "3.12"]
+
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v4
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Install dependencies
+        run: |
+          pip install ".[dev]"
+      - name: Test with pytest
+        run: |
+          pytest --color=yes -v tests
+
+  deploy:
+    name: Deploy
+    # TODO: Add back successful test requirement once package is more stable
+    # needs: test
+    # if: success() && startsWith(github.ref, 'refs/tags/') && github.event_name != 'schedule'
+    if: startsWith(github.ref, 'refs/tags/') && github.event_name != 'schedule'
```

Comment:
i think you can uncomment this.  it's better to be annoyed by a failing test blocking a release than to accidentally release a broken package (since you can never reuse that version number again).  also... tests seem to be pretty stable here?